python/tank/authentication/sso_saml2/core/sso_saml2_core.py

```diff
@@ -1015,9 +1015,22 @@ def on_dialog_closed(self, result):
                 # Let's have another go, without any cookies this time !
                 # This will force the GUI to be shown to the user.
                 self._logger.debug(
-                    "Unable to login/renew claims automaticall, presenting GUI to user"
+                    "Unable to login/renew claims automatically, presenting "
+                    "GUI to user"
                 )
-                status = self.on_sso_login_attempt()
+                if self._is_executable_nuke:
```

Comment:
I do not like to have application-specific code in tk-core.

This should be done in the nuke engine, patching the tk-core after it has been imported.